tests/res_diff_expect/fail_revisions_with_diff_list.ion

```diff
@@ -0,0 +1,17 @@
+$ion_1_0
+{
+ test: {
+  read_compare: {
+   message: "Read_compare: two revisions agree with each other but have different disagree lists. ",
+   disagree_lists: {
+    'ion-java_1': [
+     "ion-c_1",
+     "ion-js_1"
+    ],
+    'ion-java_2': [
+     "ion-c_1"
+    ]
```

Comment:
Sure, 

but in case I misunderstand it, I guess they should have the same disagree list as long as two implementations agree with each other right? 

If impl_1 agrees with impl_2 and impl_1 disagrees with impl_3. then impl_2 has to disagree with impl_3 right?

Reply:
The goal is to have no disagreements between implementations. So if impl_1 disagrees with impl_3, but impl_2 agrees with impl_3 and does not disagree with any additional implementations, then that is a step in the right direction. We should still include the difference in the results, but we should not fail the analysis because of this.

In other words, if impl_2's disagree list is a strict subset of impl_1's disagree list, then that is a positive change.

Reply:
Got it, changed the logic so it only alerts users when it going the worse direction. This test renamed as `'pass_revisions_with_diff_list.ion'` now.

I also added another test case `'fail_revisions_with_diff_list.ion'` which went the incorrect direction and start disagrees other implementations, which returns non-zero and alert users.